command/agent/auth/auth.go

```diff
@@ -74,9 +75,12 @@ func backoffOrQuit(ctx context.Context, backoff time.Duration) {
 	}
 }
 
-func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) {
+func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod, errCh chan<- error) {
 	if am == nil {
-		panic("nil auth method")
+		err := errors.New("nil auth method")
+		ah.logger.Error("auth handler internal error", "error", err)
```

Comment:
Emitting the error in `AgentCommand.Run` means that you lose the context around who emitted the error since loggers are named and the initial description in the log message can provide richer information around where the error happened. 

I wonder if it's better to simply not call `c.UI.Error` in `AgentCommand`.

Reply:
I wanted to re-visit this, and I think it makes sense for fatal errors (i.e. errors that would cause the program to terminate) to be emitted directly out for the UI output. This mainly affects template's `Server.Run` log output on previous errors that were being emitted through its logger but no longer do so, but this might be fine.